src/Lecture2.hs

```diff
@@ -162,9 +181,53 @@ data Knight = Knight
     { knightHealth    :: Int
     , knightAttack    :: Int
     , knightEndurance :: Int
-    }
-
-dragonFight = error "TODO"
+    } deriving (Show)
+
+data Treasure = Treasure
+    { content :: String
+    } deriving (Show)
+
+data Chest = Chest
+    { treasure :: Maybe Treasure
+    , gold     :: Int
+    } deriving (Show)
```

Comment:
You can make treasure polymorphic by introducing the type variable in the data type definition and using it inside the `treasure` field:
```suggestion
data Chest a = Chest
    { treasure :: Maybe a
    , gold     :: Int
    } deriving (Show)
```